pywbem/_logging.py

```diff
@@ -14,371 +14,303 @@
-extends the python logging facility to include the log_detail parameter which
-defines whether all the information defined or only a limited size is output.
-This is used with pywbem because the logs on operation and http responses can be
-very large.
-
-The code that executes the loggers call the function get_logger(..) to get a
-logger from a defined logger name.  If that logger has not yet been defined
-in PywbemLoggers, an entry will be added with the default parameters.
-"""
-from __future__ import absolute_import
+Pywbem logging is used to record information passing between the pywbem client
+and WBEM servers but not as a general recorder for errors, state, etc. within
+pywbem. In effect it is a trace tool.
+Pywbem errors are generally passed to the pywbem API user as Python
+exceptions rather than being recorded in a log by a pywbem logger.
 
-import logging
-import six
+Pywbem logging supports two :class:`~py:logging.Logger` objects
+(named loggers):
 
-__all__ = ['PywbemLoggers',
-           'LOG_OPS_CALLS_NAME', 'LOG_HTTP_NAME', 'LOG_COMPONENTS',
-           'LOG_DESTINATIONS', 'LOG_DETAIL_LEVELS', 'DEFAULT_LOG_DETAIL_LEVEL',
-           'DEFAULT_MAX_LOG_ENTRY_SIZE', 'DEFAULT_LOG_DESTINATION']
+* ``pywbem.api`` - logs user-issued calls to :class:`~pywbem:WBEMConnection`
+  methods that drive WBEM operations (see :ref:`WBEM operations`) and the
+  responses before they are passed back to the user. Log entries are defined at
+  DEBUG log level. This logs the parameters of each request and the results
+  including CIMObjects/exceptions in each method repoonse.
 
-#: Name of logger for logging user-issued calls to
-#: :class:`~pywbem.WBEMConnection` methods that drive WBEM operations.
-LOG_OPS_CALLS_NAME = 'pywbem.ops'
+* ``pywbem.http`` - logs HTTP requests and responses between the pywbem client
+  and WBEM server. This logs the http request data and response data including
+  http headers and XML payload.
 
-#: Name of logger for HTTP requests and responses between the pywbem client and
-#: WBEM servers.
-LOG_HTTP_NAME = 'pywbem.http'
 
-#: List of the logger names that :class:`~pywbem._logging.PywbemLoggers`
-#: recognizes.
-LOG_COMPONENTS = ['ops', 'http', 'all']
+Both of the above loggers define log entries when logging is at the DEBUG log
+level.  Also, when either logger is active, each new WBEMConnection is logged.
 
-#: List of log destinations that :class:`~pywbem._logging.PywbemLoggers`
-#: recognizes.
-LOG_DESTINATIONS = ['file', 'stderr', 'none']
+There are three parts to configuration of the pywbem loggers:
 
-#: List of the log detail strings that :class:`~pywbem._logging.PywbemLoggers`
-#: recognizes.
-LOG_DETAIL_LEVELS = ['all', 'min']
+* **Configure the loggers** - This involves setting the python logging
+  parameters that control the logger and its output and may be done by either
+  python logging methods or methods within WBEMConnection. It requires defining
```

Comment:
Apparently, using Python logging methods does activate the loggers, so how can they be used instead of the WBEMConnection methods?

Reply:
It configures the loggers.  They are only activated when connected to a specific WBEMConnection or to all connections. I am trying to say this without talking about putting the LogOperationRecorder into a list.